mongodb/6.0/modules/ROOT/pages/mongodb-connector-upgrade-migrate.adoc

```diff
@@ -191,6 +191,38 @@ This operation supports pagination. Each item returned as a JSON type.
 
 |===
 
+== Changes in Date Handling When Migrating from Mule 3 to Mule 4
+
+When migrating applications from Mule 3 to Mule 4 using the MongoDB connector, it is crucial to consider the following changes in date handling:
+
+* **Automatic Conversion to UTC:**
+  In Mule 3, the MongoDB connector stored dates exactly as they were sent, preserving the original time zone. However, in Mule 4, all dates are automatically converted to UTC before being stored in MongoDB. This change ensures more consistent and standardized date handling in a distributed environment.
+
+* **ISO 8601 Format:**
+  Dates in JSON documents inserted or retrieved from MongoDB are transformed into the ISO 8601 format. This includes adjusting the time value to reflect the UTC conversion.
+
```

Comment:
The only thing missing is that all string values that match the ISO 8601 format will be transformed and stored as Date in MongoDB, see W-15617168